object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/BooleanFilterNormalizer.java

```diff
@@ -176,6 +192,134 @@ private int comparePrimaryPredicateExpr(boolean isFirstNegated, PrimaryPredicate
          return -1;
       }
 
+      private void optimizeOverlappingIntervalPredicates(List<BooleanExpr> children, boolean isConjunction) {
+         for (int i = 0; i < children.size(); i++) {
+            BooleanExpr ci = children.get(i);
+            if (ci instanceof ComparisonExpr) {
+               ComparisonExpr first = (ComparisonExpr) ci;
+               assert first.getLeftChild() instanceof PropertyValueExpr;
+               assert first.getRightChild() instanceof ConstantValueExpr;
```

Comment:
Comparing field with field or constant with constant is not supported. But that type of jpa query cannot be built by the DSL builder, so we are safe.